### PR TITLE
CBAPI-4995: Update SDK to use POST search validation instead of GET	

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-4995

## Pull Request Description

Change all the GET calls on /api/investigate/v1/orgs/{orgkey}/processes/search_validation to the newer POST calls on /api/investigate/v2/orgs/{orgkey}/processes/search_validation.  Also fixed up docstrings in the process code to better match the docstring spec.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Tests have been modified to pass with the new call.  Process search has been executed against `prod02`, mainly to test a new example being added to the docstrings.

### CODE REVIEW

**Review by mrangelov-cb — APPROVED** [review]

**Review by emitreva-cb — APPROVED** [review]

File: src/cbc_sdk/platform/processes.py [hunk 1]
```diff
@@ -29,29 +48,33 @@
 
 
 class Process(UnrefreshableModel):
-    """Represents a process retrieved by one of the Enterprise EDR endpoints.
+    """
+    Information about a process running on one of the endpoin ts connected to the Carbon Black Cloud.
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/platform/processes.py:52]: Typo endpoints

File: src/cbc_sdk/platform/processes.py [hunk 2]
```diff
@@ -29,29 +48,33 @@
...
+
+    Objects of this type are retrieved through queries to the Carbon Black Cloud server, such as via
+    ``AsyncProcessQuery``.
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/platform/processes.py:55]: Seems odd to mention AsyncProcessQuery then have an example using synchronous query

File: src/cbc_sdk/platform/processes.py [hunk 3]
```diff
@@ -947,12 +988,23 @@ def _search(self, start=0, rows=0):
...
+            org.search.events(CREATE, READ)
+        """
         for item in self.results:
             yield item
 
     @property
     def results(self):
-        """Save query results to self._results with self._search() method."""
+        """
+        Save query results to ``self._results`` with ``self._search()`` method.
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/platform/processes.py:1003]: Seems weird to reference protected functions in the public facing results function

File: src/cbc_sdk/platform/processes.py (hunk 1 above)
Comment (abowersox-cb) [reply src/cbc_sdk/platform/processes.py:52]: Doyyyyyyy! 🤦‍♀️ Fixing this.

File: src/cbc_sdk/platform/processes.py (hunk 2 above)
Comment (abowersox-cb) [reply src/cbc_sdk/platform/processes.py:55]: I didn't name the class.  Despite the name, `AsyncProcessQuery` can be used synchronously. Maybe the name needs changing?

File: src/cbc_sdk/platform/processes.py (hunk 3 above)
Comment (abowersox-cb) [reply src/cbc_sdk/platform/processes.py:1003]: That whole summary line is unsatisfactory. I'm completely rewriting it.

**Review by kebringer-cb — APPROVED** [review]

**Pull request merged**